### PR TITLE
randomports: give user ability to init ipfs using random port for swarm.

### PR DESCRIPTION
**Pull request by tarekbadrsh**
Issue : https://github.com/ipfs/go-ipfs/issues/5176

It's still under development, just to make sure that the code is going to
do as we expected.

### CODE REVIEW

**Comment by tarekbadrshalaan** [comment]
Hi @Stebalien,
- I change the description.
- I don't know how do we test this repo, I can't find any test pkg.
- I have another idea about random port, we can generate random number and check if it's available, ex.
```
	src := rand.NewSource(time.Now().UnixNano())
	rnd := rand.New(src)
	portRange := 10000
	port := rnd.Intn(portRange)
	for !isAvailablePort(port) {
		port = rnd.Intn(portRange)
	}
```
please feel free to ask about more update.

**Comment by djdv** [comment]
@tarekbadrshalaan 
>I don't know how do we test this repo, I can't find any test pkg.

There are tests in `go-ipfs` at the moment that check the output of the config command.
https://github.com/ipfs/go-ipfs/blob/master/test/sharness/t0021-config.sh
It may be best to add your cases there.

>I have another idea about random port, we can generate random number and check if it's available, ex.

Is there benefit to this over the randomness inherited from [`net.Listen("tcp", "[::]:0")`](https://github.com/ipfs/go-ipfs-config/blob/196c87c91ce76bbbca78b6a35c0fbe7e5f524e4e/profile.go#L184)?

File: profile.go [hunk 1]
```diff
@@ -159,6 +163,31 @@ fetching may be degraded.
 			return nil
 		},
 	},
+	"randomports": {
+		Description: `Use a random port number for swarm.`,
```
Comment (tarekbadrshalaan) [inline profile.go]: Hello @djdv  
as you can see here I updated the discretion as you said.

again sorry be late and please feel free to ask about more update.

**Comment by tarekbadrshalaan** [comment]
> Is there benefit to this over the randomness inherited from [`net.Listen("tcp", "[::]:0")`](https://github.com/ipfs/go-ipfs-config/blob/196c87c91ce76bbbca78b6a35c0fbe7e5f524e4e/profile.go#L184)?
 * no it was just an idea i thought it might help.

**Comment by djdv** [comment]
@tarekbadrshalaan 
Sorry for my absence, something unexpected came up.
The delay is on me this time ;^)

Patch seems good to me, handing off to @Stebalien

**Pull request merged**